src/components/primitive/Typography/typography.component.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import styled from 'styled-components';
+import { Base, Title, ContentHeader, Small, Content, Description } from './typography.component.styles';
+import { TextPresets, TextProps } from './typography.component.props';
+
+const Text: React.FC<TextProps> = ({ size, color, lineHeight, upperCase, variant, children }) => {
+  let TextComponent;
+
+  switch (variant) {
+    case TextPresets.TITLE:
+      TextComponent = Title;
+      break;
+    case TextPresets.CONTENT_HEADER:
+      TextComponent = ContentHeader;
+      break;
+    case TextPresets.CONTENT:
+      TextComponent = Content;
+      break;
+    case TextPresets.DESCRIPTION:
+      TextComponent = Description;
+      break;
+
+    case TextPresets.SMALL:
+      TextComponent = Small;
+      break;
+
+    default:
+      TextComponent = Base;
+  }
+
+  const Typography = styled(TextComponent)`
+    /* custom styles goes here */
+
+    ${({ center }) => (center ? `text-align: center` : '')};
+    ${({ lineHeight }) => (lineHeight ? `line-height: ${lineHeight}px` : '')};
+
+    ${({ upperCase }) => (upperCase ? `text-transform: uppercase` : '')};
+    ${({ lowerCase }) => (lowerCase ? `text-transform: lowercase` : '')};
+    ${({ paddingLeft }) => (paddingLeft ? `padding-left: ${paddingLeft}px` : '')};
```

Comment:
Should we go with px? How would affect the responsiveness?

Reply:
replaced this with a rem unit.